flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala

```diff
@@ -455,13 +465,15 @@ object GenerateUtils {
   def generateRowtimeAccess(
       ctx: CodeGeneratorContext,
       contextTerm: String): GeneratedExpression = {
+    val resultType = new TimestampType(true, TimestampKind.ROWTIME, 3)
```

Comment:
IMO rowtime and proctime should be Timestamp(3) and keep millisecond precision.
Is it necessary to support higher precision rowtime and proctime?

Reply:
If user declares a field with TIMESTAMP(9) and tells framework that field is row time attribute, we still have to support it, right?
For proctime, since it's generated by framework, I'm ok with treating it as TIMESTAMP(3)

Reply:
This is used for `STREAMRECORD_TIMESTAMP` and `MATCH_ROWTIME`. The first one is used to access to timestamp of a `StreamRecord` which is a long value. The second one is used to access an event time attribute from MATCH_RECOGNIZE which is also a long value.
They use `ProcessFunction.Context`.`timestamp()` to get the underlying long value.

I will change this function name to `generateTimestampAccess` to avoid confusing.